components/feature/prompts/src/androidTest/java/mozilla/components/feature/prompts/file/OnDeviceFilePickerTest.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.file
+
+import android.content.ClipData
+import android.content.Context
+import android.content.Intent
+import androidx.core.net.toUri
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.feature.prompts.PromptContainer
+import org.junit.Assert
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class OnDeviceFilePickerTest {
+    private val context: Context
+        get() = ApplicationProvider.getApplicationContext()
+
+    private val badUri1 = "file:///data/user/0/${context.packageName}/any_directory/file.text".toUri()
+    private val badUri2 = "file:///data/data/${context.packageName}/any_directory/file.text".toUri()
+    private val goodUri = "file:///data/directory/${context.packageName}/any_directory/file.text".toUri()
+
+    @Test
+    fun filterUrisUnderPrivateAppDirectory() {
+        val uris = arrayOf(badUri1, badUri2, goodUri)
+
+        val filterUris = uris.filterUrisUnderPrivateAppDirectory(context)
+
+        Assert.assertEquals(1, filterUris.size)
+        assertTrue(filterUris.contains(goodUri))
+    }
+
+    @Test
+    fun unsafeUrisWillNotBeSelected() {
+        val promptContainer = PromptContainer.TestPromptContainer(context)
+
+        val filePicker = FilePicker(promptContainer, BrowserStore()) { }
+        var onDismissWasExecuted = false
+
+        val filePickerRequest = PromptRequest.File(
```

Comment:
Maybe it would be good have dismiss/selected asserts in all tests? - have all pickers constructed the same and then do all the same asserts, with only one difference depending on the Intent.
Just to make sure some later refactorization doesn't silently break something?